src/cli/java/org/commcare/util/screen/SyncScreen.java

```diff
@@ -129,6 +103,13 @@ public boolean prompt(PrintStream printStream) throws CommCareSessionException {
     @Override
     public boolean handleInputAndUpdateSession(CommCareSession commCareSession, String s, boolean allowAutoLaunch) throws CommCareSessionException {
         if (syncSuccessful) {
+            updateSessionOnSuccess();
+            Entry commandEntry = sessionWrapper.getCurrentEntry();
```

Comment:
@shubham1g5 with reference to this comment here: https://github.com/dimagi/commcare-core/pull/1109#discussion_r870999599, do you think this needs to use the passed in session rather than the one in the screen?

`getCurrentEntry` callsed `getCommand` on the session so it relies on the command in the session being correct. I THINK that the session the screen is initialised with will have the correct command.

Reply:
I think it's ok to use the passed session object to perform any READ operations as it tend to give the same result as using the screen's copy of session in almost all cases, Though the safest thing to do will be to rely on the passed instance of session just in case it was updated before this method was called.